ansible/roles/test/tasks/add_container_to_inventory.yml

```diff
@@ -0,0 +1,24 @@
+- name: Add/update the public key in the "{{ lookup('env','HOME') + '/.ssh/known_hosts' }}"
+  delegate_to: localhost
+  known_hosts:
+    name: "{{ ansible_host }}"
+    key: "{{ lookup('pipe', 'ssh-keyscan ' + ansible_host) }}"
+    state: "present"
+
+- name: Generate an OpenSSH keypair
+  delegate_to: localhost
+  openssh_keypair:
+    path: "{{ lookup('env','HOME') + '/.ssh/id_rsa' }}"
+
+- name: Set authorized key
+  authorized_key:
+    user: "{{ ansible_ssh_user }}"
+    state: present
+    key: "{{ lookup('file', lookup('env','HOME') + '/.ssh/id_rsa.pub') }}"
```

Comment:
>key [](start = 4, length = 3)

You can get `key` from previous `openssh_keypair` return values.

Reply:
Thanks for your refinement! I feel `public_key` is just more straightforward.

---
In reply to: [356376796](https://github.com/Azure/sonic-mgmt/pull/1269#discussion_r356376796) [](ancestors = 356376796)

Reply:
I do not choose the name of the public key file. 

"Name of the files containing the public and private key. The file containing the public key will have the extension .pub. "
https://docs.ansible.com/ansible/latest/modules/openssh_keypair_module.html#parameters

Reply:
Maybe there is some miscommunication. I suggest
```
    key: out.public_key
```
Could you help explorer this one? Seems it is designed just for this purpose.

---
In reply to: [356776386](https://github.com/Azure/sonic-mgmt/pull/1269#discussion_r356776386) [](ancestors = 356776386,356376796)

Reply:
Oh, yes, I missed this option. Changed.